applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp

```diff
@@ -187,15 +187,33 @@ namespace Kratos {
         ProcessInfo& r_process_info = r_model_part.GetProcessInfo();
 
         if (r_process_info[SEARCH_CONTROL] == 0) {
-            for (int i = 0; i < mNumberOfThreads; i++) {
-                if (r_process_info[SEARCH_CONTROL_VECTOR][i] == 1) {
-                    r_process_info[SEARCH_CONTROL] = 1;
-                    if(r_model_part.GetCommunicator().MyPID() == 0) {
-                        KRATOS_WARNING("DEM") << "From now on, the search is activated because some failure occurred " << std::endl;
+
+            ElementsArrayType& pElements = r_model_part.GetCommunicator().LocalMesh().Elements();
+            bool some_bond_is_broken = false;
+
+            #pragma omp parallel for
+            for (int i= 0; i < (int)pElements.size(); i++) {
+
+                ElementsArrayType::iterator it = pElements.ptr_begin() + i;
+                Element* p_element = &(*it);
+                SphericContinuumParticle* p_sphere = dynamic_cast<SphericContinuumParticle*>(p_element);
+
+                for (int j=0; j<(int) p_sphere->mContinuumInitialNeighborsSize; j++) {
+
+                    if (p_sphere->mIniNeighbourFailureId[j] != 0) {
+                        #pragma omp atomic write
+                        some_bond_is_broken = true;
+
+                        break;
```

Comment:
I would use the parallel utilities and the atomic utiltites here

Reply:
I just pushed this.